presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/MergeWindows.java

```diff
@@ -85,30 +85,45 @@ protected PlanNode visitPlan(
                 RewriteContext<Multimap<WindowNode.Specification, WindowNode>> context)
```

Comment:
is there a way to check which window groups are independent?

Lets take a look at com.facebook.presto.sql.planner.optimizations.joins.JoinGraph. What if you would have a directed graph where vertex is a windowNode and directed edge would point which window node depends on which window node. 

1. Then you could merge all the independent (disjoint) with the same window specification nodes into a single one.
2. Then you could browse the graph in order to span it. you could take deepest leafs and treat them as independent list of nodes which could be merged  as in the 1 step.
3. If you remove nodes which are merged from the graph you could go back to step 2 until all the nodes are processed.

Note that in first iteration of 2, nodes can be merged together with nodes from 1.

It would solve the example mentioned in the commit message, however I am not sure if it would always produce optimal plan.

I am ok to solve it as separate PR.